docs/content-services/components/infinite-scroll-datasource.md

```diff
@@ -0,0 +1,66 @@
+---
+Title: Infinite Scroll Datasource
+Added: v6.6.0
+Status: Active
+Last reviewed: 2024-01-15
+---
+
+# [Infinite Scroll Datasource](../../../lib/content-services/src/lib/infinite-scroll-datasource/infinite-scroll-datasource.ts "Defined in infinite-scroll-datasource.ts")
+
+Contains abstract class acting as a baseline for various datasources for infinite scrolls. 
+
+## Basic Usage
+
+First step to use infinite scroll datasource in any component is creating a datasource class extending `InfiniteScrollDatasource` using specific source of data e.g. one of the content endpoints.
+```ts
+export class VersionListDataSource extends InfiniteScrollDatasource<VersionEntry> {
+    constructor(private versionsApi: VersionsApi, private node: Node) {
+        super();
+    }
+
+    getNextBatch(pagingOptions: ContentPagingQuery): Observable<VersionEntry[]> {
+        return from(this.versionsApi.listVersionHistory(this.node.id, pagingOptions)).pipe(
+            take(1),
+            map((versionPaging) => versionPaging.list.entries)
+        );
+    }
+}
+```
+
+Then in component that will have the infinite scroll define the datasource as instance of a class created in previous step, optionally you can set custom size of the items batch or listen to loading state changes:
+```ts
+ this.versionsDataSource = new VersionListDataSource(this.versionsApi, this.node);
```

Comment:
redundant empty space before this line (it causes not aligned code)

Reply:
Removed